collections/forms/src/validators/__tests__/dhis2Username.test.js

```diff
@@ -21,33 +21,37 @@ describe('validator: dhis2Username', () => {
         ])
     })
 
-    describe('does not allow usernames to start with _, . or @', () => {
+    describe('does not allow usernames to start with _, -, . or @', () => {
         testValidatorValues(dhis2Username, invalidUsernameMessage, [
             '_xxx',
+            '-xxx',
             '.xxx',
             '@xxx',
         ])
     })
 
-    describe('does not allow usernames to end with _, . or @', () => {
+    describe('does not allow usernames to end with _, -, . or @', () => {
         testValidatorValues(dhis2Username, invalidUsernameMessage, [
             'xxx_',
+            'xxx-',
             'xxx.',
             'xxx@',
         ])
     })
 
-    describe('does not allow usernames to contain __, .. or @@', () => {
+    describe('does not allow usernames to contain __, --, .. or @@', () => {
         testValidatorValues(dhis2Username, invalidUsernameMessage, [
-            '__xx',
-            '..xx',
-            '@@xx',
+            'xx__xx',
+            'xx--xx',
+            'xx..xx',
+            'xx@@xx',
```

Comment:
I decided to update these test values because the previous values didn't just contain repeating disallowed characters, but they also started with these characters. As such these values are invalid for 2 reasons, they would have been rejected due to to the _does not allow usernames to start with ..._ pattern as well. By having these characters in the middle of the string we are ensuring there is only a single violation of the pattern in play.